### ISSUE TITLE
Flutter Wrapper does not download new flutter versions on CI

### ISSUE THREAD

**Issue by knaeckeKami** [passsy/flutter_wrapper#16]
After updatin flutter in my local repository, I expected flutterw to download flutter also on the CI.
This does not happen for my configuration, I get the following errors when building on the CI:
```
$ ./flutterw clean
Warning: Detected detached HEAD. '.flutter' submodule points to a specific commit, not a branch.
You can ignore this warning when you run './flutterw' for the first time after cloning the repository.
error: branch 'dev' not found.
fatal: 'dev' could be both a local file and a tracking branch.
Please use -- (and optionally --no-guess) to disambiguate
Warning: Commit 1946fc4da0f80c522d7e3ae7d4f7309908ed86f2 in submodule '.flutter' is not part of the branch 'dev' as defined in .gitmodules
You have to manually switch to a valid branch (i.e. './flutterw channel stable') or './flutterw upgrade' will fail
Alternatively change the branch 'dev' in .gitmodules to the correct upstream branch (one of stable|beta|dev).
```
My .gitmodules file:

```
[submodule ".flutter"]
	path = .flutter
	url = https://github.com/flutter/flutter.git
	branch = dev
```

The CI is Gitlab CI.

**Comment by passsy** [comment]
How did you upgrade? Have you actually run `./flutterw upgrade`? It looks to me like you manually edited the `.gitmodules` file.
Also make sure you committed the submodule change, not only committed `.gitmodules`

**Comment by knaeckeKami** [comment]
I did not manually edit git .gitmodules file, and I committed the submodule change. I ran ./flutterw upgrade.

The CI is able to pick and download the correct flutter version for the first time, but subsequent changes fail.
If I remove flutterw and add it again, it works. 
But changing the flutter version with `./flutterw channel dev` and` ./flutterw upgrade` does not work, it only changes the flutter version on my machine, but the CI fails with the error that I posted.

It might be also be a quirk of Gitlab CI.
